src/Upload.tsx

```diff
@@ -0,0 +1,129 @@
+        const { t } = useTranslation();
+        const {
+            className,
+            disabled = false,
+            hint = t("hint"),
+            multiple = false,
+            state = "default",
+            stateRelatedMessage,
+            nativeInputProps = {}
+        } = props;
+
+        const inputId = (function useClosure() {
+            const id = useId();
+
+            return nativeInputProps.id ?? `input-${id}`;
+        })();
+
+        const messageId = `${inputId}-desc-error`;
+        return (
+            <div
+                className={cx(
+                    fr.cx(
+                        "fr-upload-group",
+                        disabled && "fr-input-group--disabled",
+                        (() => {
+                            switch (state) {
+                                case "error":
+                                    return "fr-input-group--error";
+                                case "success":
+                                    return "fr-input-group--valid";
+                                case "default":
+                                    return undefined;
+                            }
+                            assert<Equals<typeof state, never>>(false);
+                        })()
+                    ),
+                    className
+                )}
+                ref={ref}
+            >
+                <label className={fr.cx("fr-label")} aria-disabled={disabled} htmlFor={inputId}>
+                    {t("add_files")}
+                    <span className={fr.cx("fr-hint-text")}>{hint}</span>
+                </label>
+                <input
+                    aria-describedby={messageId}
+                    aria-disabled={disabled}
+                    className={cx(fr.cx("fr-upload"))}
+                    disabled={disabled}
+                    id={inputId}
+                    multiple={multiple}
+                    name={inputId}
+                    type="file"
+                />
+                {state !== "default" && (
+                    <p
+                        id={messageId}
+                        className={cx(
+                            fr.cx(
+                                (() => {
+                                    switch (state) {
+                                        case "error":
+                                            return "fr-error-text";
+                                        case "success":
+                                            return "fr-valid-text";
+                                    }
+                                    assert<Equals<typeof state, never>>(false);
+                                })()
+                            )
+                        )}
+                    >
+                        {stateRelatedMessage}
+                    </p>
+                )}
+            </div>
+        );
+    })
+);
+
+Upload.displayName = "Upload";
```

Comment:
```suggestion
Upload.displayName = symToStr({ Upload });
```
Without this, if the component ever gets rename you can be damn sure that the Display name won't be.   
We want to ensure concistency betwen how componens are named, their display name and the functions to add translations.